src/mesido/workflows/gas_elect_workflow.py

```diff
@@ -0,0 +1,244 @@
+    TechnoEconomicMixin,
+    LinearizedOrderGoalProgrammingMixin,
+    SinglePassGoalProgrammingMixin,
+    ESDLMixin,
+    CollocatedIntegratedOptimizationProblem,
+):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._number_of_years = 1.0
+
+        # variables for solver settings
+        self._qpsol = CachingQPSol()
+
+        self._save_json = True
+
+    def energy_system_options(self):
+        options = super().energy_system_options()
+        options["neglect_pipe_heat_losses"] = True
+        options["include_electric_cable_power_loss"] = False
+        # TODO: determine why no heat pump (case with heat pumps & boilers) is used when pwer
+        # losses are included
+        # options["include_electric_cable_power_loss"] = True
+
+        # Setting when started with head loss inclusions
+        self.gas_network_settings["minimum_velocity"] = 0.0
+        self.gas_network_settings["maximum_velocity"] = 15.0
+
+        # TODO: resolve scaling and potential other issues preventing HIGHS to optimize the system
+        # when LINEARIZED_N_LINES_EQUALITY head loss setting is used
+        # self.gas_network_settings["n_linearization_lines"] = 3
+        # self.gas_network_settings["minimize_head_losses"] = False
+        # self.gas_network_settings["head_loss_option"] = HeadLossOption.LINEARIZED_N_LINES_EQUALITY
+
+        self.gas_network_settings["minimize_head_losses"] = True
+        self.gas_network_settings["head_loss_option"] = HeadLossOption.LINEARIZED_ONE_LINE_EQUALITY
+
+        return options
+
+    def solver_options(self):
+        options = super().solver_options()
+        options["casadi_solver"] = self._qpsol
+        options["solver"] = "highs"
+        highs_options = options["highs"] = {}
+        highs_options["mip_abs_gap"] = 1.0e-6
+
+        return options
+
+    def pre(self):
+        super().pre()
+
+        self._qpsol = CachingQPSol()
+
+        # Convert gas demand Nm3/h (data in timeseries source file) to heat demand in watts
+        # Assumumption:
+        #   - gas heating value (LCV value) = 31.68 * 10^6 (J/m3) at 1bar, 273.15K
+        #   - gas boiler efficiency 80%
+        # TODO: setup a standard way for gas usage and automate the link to heating value & boiler
+        # efficiency (if needed)
+        for demand in self.energy_system_components["heat_demand"]:
+            target = self.get_timeseries(f"{demand}.target_heat_demand")
+
+            # Manually set heating demand values
+            boiler_efficiency = 0.8
+            gas_heating_value_joule_m3 = 31.68 * 10**6
+            for ii in range(len(target.values)):
+                target.values[ii] *= gas_heating_value_joule_m3 * boiler_efficiency
+
+            self.io.set_timeseries(
+                f"{demand}.target_heat_demand",
+                self.io._DataStore__timeseries_datetimes,
+                target.values,
+                0,
+            )
```

Comment:
why is this in the pre? Normally we tweak timeseries in the the read method.

Reply:
Code has been updated. As a side note, this was also done in pre() in run_hydraulic_power.py